docs/customization/model.rst

```diff
@@ -148,44 +167,61 @@ Just like for regular models you can also check the class of translatable models
 
     $ php bin/console debug:container --parameter=sylius.model.shipping_method.class
 
-**1.** The first thing to do is to write your own class which will extend the base ``ShippingMethod`` class.
+**1.** The first thing to do is to add your own interface ``App\Entity\Shipping\ShippingMethodInterface`` which will extend the base ``Sylius\Component\Core\Model\ShippingMethodInterface`` interface.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Entity\Shipping;
+
+    use Sylius\Component\Core\Model\ShippingMethodInterface as BaseShippingMethodInterface;
+
+    interface ShippingMethodInterface extends BaseShippingMethodInterface
+    {
+        public function getEstimatedDeliveryTime(): ?string;
+
+        public function setEstimatedDeliveryTime(?string $estimatedDeliveryTime): void;
+    }
+
+**2.** Next thing to do is to add your own fields in class ``App\Entity\Shipping\ShippingMethod`` which will extend the base ``Sylius\Component\Core\Model\ShippingMethod`` class.
+
+File ``App\Entity\Shipping\ShippingMethod`` existing in Sylius-Standard. We need only modificate it.
 
 .. code-block:: php
 
     <?php
 
-    namespace App\Entity;
+    declare(strict_types=1);
+
+    namespace App\Entity\Shipping;
 
+    use Doctrine\ORM\Mapping as ORM;
     use Sylius\Component\Core\Model\ShippingMethod as BaseShippingMethod;
-    use Sylius\Component\Shipping\Model\ShippingMethodTranslation;
+    use Sylius\Component\Shipping\Model\ShippingMethodTranslationInterface;
 
-    class ShippingMethod extends BaseShippingMethod
+    /**
+     * @ORM\MappedSuperclass()
+     * @ORM\Table(name="sylius_shipping_method")
+     */
+    class ShippingMethod extends BaseShippingMethod implements ShippingMethodInterface
     {
-        /**
-         * @var string
-         */
+        /** @ORM\Column(type="string",nullable=true) */
         private $estimatedDeliveryTime;
 
-        /**
-         * @return string
-         */
-        public function getEstimatedDeliveryTime(): string
+        public function getEstimatedDeliveryTime(): ?string
         {
             return $this->estimatedDeliveryTime;
         }
 
-        /**
-         * @param string $estimatedDeliveryTime
-         */
-        public function setEstimatedDeliveryTime(string $estimatedDeliveryTime): void
+        public function setEstimatedDeliveryTime(?string $estimatedDeliveryTime): void
         {
             $this->estimatedDeliveryTime = $estimatedDeliveryTime;
         }
 
-        /**
-         * {@inheritdoc}
-         */
-        protected function createTranslation(): ShippingMethodTranslation
+        protected function createTranslation(): ShippingMethodTranslationInterface
```

Comment:
Is this method needed here? What happens if it isn't customised?

Reply:
We got error because we need use Interface now, probably using class is obsolete

Reply:
If we delete createTranslation() we also got error